### PR TITLE
Allow per-request base URIs in HttpClient

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

File: api/client/src/main/java/org/projectnessie/client/http/HttpRequest.java [hunk 1]
```diff
@@ -31,7 +35,16 @@ public abstract class HttpRequest
...
 
   protected HttpRequest(HttpRuntimeConfig config) {
-    this.uriBuilder = new UriBuilder(config.getBaseUri());
+    this(config, config.getBaseUri());
+  }
+
+  protected HttpRequest(HttpRuntimeConfig config, URI baseUri) {
+    requireNonNull(baseUri, "Base URI cannot be null");
+    checkArgument(
+        "http".equals(baseUri.getScheme()) || "https".equals(baseUri.getScheme()),
```
Comment (dimas-b) [inline api/client/src/main/java/org/projectnessie/client/http/HttpRequest.java:44]: Maybe `equalsIgnoreCase`? Per [rfc3986](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) implementations should accept either case.

**Review by dimas-b — COMMENTED** [review]
LGTM :+1:

File: api/client/src/main/java/org/projectnessie/client/http/HttpClient.java [hunk 2]
```diff
@@ -38,12 +38,12 @@ enum Method {
 
   HttpRequest newRequest();
 
+  HttpRequest newRequest(URI baseUri);
+
   static Builder builder() {
     return new HttpClientBuilderImpl();
   }
 
-  URI getBaseUri();
```
Comment (snazy) [inline api/client/src/main/java/org/projectnessie/client/http/HttpClient.java:45]: Please leave this one

File: .github/actions/ci-incr-build-cache-prepare/action.yml [hunk 3]
```diff
@@ -31,16 +31,18 @@ runs:
     - name: Gradle / Init
       uses: gradle/gradle-build-action@v2
       with:
-        cache-read-only: ${{ inputs.cache-read-only }}
+#        cache-read-only: ${{ inputs.cache-read-only }}
```
[the remark this answers is not in the document]
Comment (adutra) [reply .github/actions/ci-incr-build-cache-prepare/action.yml:34]: Trying to fix gradle build cache issues.

**Comment by adutra** [comment]
Authentication is another component that could benefit from a per-request override mechanism – I can look into that in another PR.

**Review by dimas-b — APPROVED** [review]

**Pull request merged**